src/components/Header/Header.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { postTodo } from '../../api/todos';
+import { formatTodo } from '../../utils/formatResponse';
+
+type Props = {
+  todosToRender: Todo[];
+  setTodosToRender: (arr: Todo[]) => void;
+  pushError: (title: string) => void;
+  setTempTodo: (todo: Todo | null) => void;
+  setIsTempLoading: (val: boolean) => void;
+  setTodos: (arr: Todo[]) => void;
+};
+
+export const Header: React.FC<Props> = ({
+  todosToRender,
+  setTodosToRender,
+  pushError,
+  setTempTodo,
+  setIsTempLoading,
+  setTodos,
+}) => {
+  const [value, setValue] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [focus, setFocus] = useState(false);
+  const refInput = useRef<HTMLInputElement>(null);
+
+  const onInputChange = (str: string) => {
+    setValue(() => str);
+  };
+
+  // this would be handled using methods later
+  const toggleAll = () => {
+    if (todosToRender) {
+      setTodosToRender([...todosToRender].map(todo => {
+        return todosToRender.every(item => item.completed)
+          ? {
+            ...todo,
+            completed: !todo.completed,
+          }
+          : {
+            ...todo,
+            completed: true,
+          };
+      }));
+    }
+  };
+
+  const onFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (!value) {
+      pushError('Title can\'t be empty');
+
+      return;
+    }
+
+    const rawTodo = {
+      title: value,
+      userId: 10222,
+      completed: false,
+    };
+
+    setIsDisabled(true);
+    setTempTodo({ id: 0, ...rawTodo });
+    setIsTempLoading(true);
+
+    postTodo(rawTodo)
+      .then(res => {
+        setTodos([...todosToRender, {
+          ...formatTodo(res),
+        }]);
+        setValue('');
+      })
+      .catch(() => {
+        pushError('Unable to add a todo');
+      })
+      .finally(() => {
+        setTempTodo(null);
+        setIsDisabled(false);
+        setFocus(true);
+      });
+  };
+
+  useEffect(() => {
+    if (focus) {
+      refInput.current?.focus();
+    }
+  }, [focus]);
+
+  return (
+    <header className="todoapp__header">
+
+      <button
+        type="button"
+        className={`todoapp__toggle-all${todosToRender?.some(todo => !todo.completed) ? ' active' : ''}`}
+        onClick={toggleAll}
+        aria-label="Toggle all"
```

Comment:
You should use 'ariaLabel' instead of 'aria-label' for better accessibility.

Reply:
<img width="510" alt="image" src="https://github.com/mate-academy/react_todo-app-add-and-delete/assets/17066980/2d0aac79-065e-48ad-be1c-420aad322482">